common/SmokeTests/SmokeTest/Update-Dependencies.ps1

```diff
@@ -0,0 +1,114 @@
+        '4.1.0-dev.20191108.1' = $true;
+        '4.1.0-dev.20191109.2' = $true;
+        '4.1.0-dev.20191110.1' = $true;
+        '4.1.0-dev.20191111.1' = $true;
+        '4.1.0-dev.20191112.1' = $true;
+        '4.1.0-dev.20191113.1' = $true;
+        '4.1.0-dev.20191114.1' = $true;
+        '4.1.0-dev.20191115.1' = $true;
+        '4.1.0-dev.20191116.1' = $true;
+        '4.1.0-dev.20191117.1' = $true;
+        '4.1.0-dev.20191118.1' = $true;
+        '4.1.0-dev.20191119.1' = $true;
+        '4.1.0-dev.20191120.1' = $true;
+        '4.1.0-dev.20191121.1' = $true;
+        '4.1.0-dev.20191122.1' = $true;
+        '4.1.0-dev.20191123.1' = $true;
+        '4.1.0-dev.20191124.1' = $true;
+        '4.1.0-dev.20191125.1' = $true;
+        '4.1.0-dev.20191127.1' = $true;
+        '4.1.0-dev.20191128.1' = $true;
+        '4.1.0-dev.20191129.1' = $true;
+        '4.1.0-dev.20191130.1' = $true;
+        '4.1.0-dev.20191201.1' = $true;
+        '4.1.0-dev.20191202.1' = $true;
+        '4.1.0-dev.20191203.1' = $true;
+        '4.1.0-dev.20191204.1' = $true;
+        '4.1.0-dev.20191205.1' = $true;
+        '4.1.0-dev.20191206.1' = $true;
+        '4.1.0-dev.20191207.1' = $true;
+        '4.1.0-dev.20191208.1' = $true;
+        '4.1.0-dev.20191209.1' = $true;
+        '4.1.0-dev.20191210.1' = $true;
+        '4.1.0-dev.20191211.1' = $true;
+        '4.1.0-dev.20191212.1' = $true;
+        '4.1.0-dev.20191213.1' = $true;
+        '4.1.0-dev.20191214.1' = $true;
+        '4.1.0-dev.20191215.1' = $true;
+        '4.1.0-dev.20191216.1' = $true;
+        '4.1.0-dev.20191217.1' = $true;
+        '4.1.0-dev.20191218.1' = $true;
+        '4.1.0-dev.20191219.1' = $true;
+        '4.1.0-dev.20191220.3' = $true;
+        '4.1.0-dev.20191221.1' = $true;
+        '4.1.0-dev.20191222.1' = $true;
+        '4.1.0-dev.20191223.1' = $true;
+        '4.1.0-dev.20191224.1' = $true;
+        '4.1.0-dev.20191225.1' = $true;
+        '4.1.0-dev.20191226.1' = $true;
+        '4.1.0-dev.20191227.1' = $true;
+        '4.1.0-dev.20191228.1' = $true;
+        '4.1.0-dev.20191229.1' = $true;
+        '4.1.0-dev.20191230.1' = $true;
+        '4.1.0-dev.20191231.1' = $true;
+        '4.1.0-dev.20200101.1' = $true;
+        '4.1.0-dev.20200102.1' = $true;
+        '4.1.0-dev.20200103.1' = $true;
+        '4.1.0-dev.20200104.1' = $true;
+        '4.1.0-dev.20200105.1' = $true;
+        '4.1.0-dev.20200106.1' = $true;
+        '4.1.0-dev.20200107.1' = $true;
+        '4.1.0-dev.20200108.1' = $true;
+        '5.0.0-dev.20190625.1' = $true;
+        '5.0.0-dev.20190626.1' = $true;
+        '5.0.0-dev.20190627.1' = $true;
+    }
+}
+
+$PACKAGE_REFERENCE_XPATH = '//Project/ItemGroup/PackageReference'
+
+# List all packages from the source specified by $FeedName. Packages are sorted
+# ascending by version according to semver rules (e.g. 4.0.0-preview.1 comes
+# before 4.0.0) not lexicographically.
+$allPackages = Find-Package -Source $FeedName -AllVersion -AllowPrereleaseVersions
```

Comment:
Isn't this going to be huge? Should we instead do this once per package name?

Reply:
The command does take a long time to run (1m-3m of a 4m-5m pipeline execution). Filtering by `-Name` is the right thing to do in normal circumstances but when working with sleet we only get back error messages and no package objects:

```
PS C:\code\azure-sdk-for-net\common\SmokeTests\SmokeTest> Find-Package -Name Azure.Security.Keyvault.Secrets -Source DevFeed
WARNING: Could not get response from query 'https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-net/virtualcatalog/data/4d3f6f1e-89b2-44d1-ab1c-fd1e31c45d2b.json'.
...
WARNING: Could not get response from query 'https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-net/virtualcatalog/data/2eda23e7-fbb3-4dc9-9c13-f3ec53c80a61.json'.
WARNING: Could not get response from query 'https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-net/virtualcatalog/data/d758c2c8-deda-4ba4-9a11-b8cd7ed6bfba.json'.
Find-Package: No match was found for the specified search criteria and package name 'Azure.Security.Keyvault.Secrets'. Try Get-PackageSource to see all available registered package sources.
```

Using a package source like nuget.org gives useful results when using `-Name`. 

I've added a comment explaining.